### ISSUE TITLE
Misc cleanup for 0.3 release

### ISSUE THREAD

**Issue by fritzo** [uber/pyro#1406]
- [x] replace `.new_tensor()` with `.detach().clone()` where possible
- [x] replace `.independent()` with `.any_less_confusing_name()` (@jpchen)
- [x] replace `.empty(...).fill_(value)` with `torch.full(shape, value)` or `torch.full_like(x, value)`
- [x] replace `len(tensor.shape)` with `tensor.dim()`
- [x] use `plate(subsample=)` in examples to get correctly scaled ELBO loss (see #1437)
- [x] rename various D_term, W_term while using LowRankMVN
- [x] add docs for `pyro.ops`
- [x] check versioning for examples
- others?

---- triaged ----
- (optional?) remove remaining uses of `unittest.TestCase`?
- replace `tensor.shape[n]` with `tensor.size(n)` (which is a tiny bit cheaper)
- ~~rename `Trace_ELBO` and friends?~~

**Comment by jpchen** [comment]
closing as most have been addressed, the rest are low priority, too much effort, or not targeted for this release.
